Sources/MessagingInApp/State/InAppMessageState.swift

```diff
@@ -0,0 +1,176 @@
+import CioInternalCommon
+import Foundation
+
+/// Represents the state of InAppMessage store.
+/// It holds all the information required to manage and identify current state of in-app messaging.
+/// It is managed by reducer and should only be updated by dispatching appropriate actions to the store.
+struct InAppMessageState: Equatable, CustomStringConvertible {
+    let siteId: String
+    let dataCenter: String
+    let environment: GistEnvironment
+    let pollInterval: Double
+    let userId: String?
+    let currentRoute: String?
+    let currentMessageState: MessageState
+    let messagesInQueue: Set<Message>
+    let shownMessageQueueIds: Set<String>
+
+    init(
+        siteId: String = "",
+        dataCenter: String = "",
+        environment: GistEnvironment = .production,
+        pollInterval: Double = 600,
+        userId: String? = nil,
+        currentRoute: String? = nil,
+        currentMessageState: MessageState = .initial,
+        messagesInQueue: Set<Message> = [],
+        shownMessageQueueIds: Set<String> = []
+    ) {
+        self.siteId = siteId
+        self.dataCenter = dataCenter
+        self.environment = environment
+        self.pollInterval = pollInterval
+        self.userId = userId
+        self.currentRoute = currentRoute
+        self.currentMessageState = currentMessageState
+        self.messagesInQueue = messagesInQueue
+        self.shownMessageQueueIds = shownMessageQueueIds
+    }
+
+    /// Copies the current state and replaces the given properties with the new values.
+    /// It is useful when updating state with only a few properties and keeping the rest as is.
+    func copy(
+        pollInterval: Double? = nil,
+        userId: String? = nil,
+        currentRoute: String? = nil,
+        currentMessageState: MessageState? = nil,
+        messagesInQueue: Set<Message>? = nil,
+        shownMessageQueueIds: Set<String>? = nil
+    ) -> InAppMessageState {
+        InAppMessageState(
+            siteId: siteId,
+            dataCenter: dataCenter,
+            environment: environment,
+            pollInterval: pollInterval ?? self.pollInterval,
+            userId: userId ?? self.userId,
+            currentRoute: currentRoute ?? self.currentRoute,
+            currentMessageState: currentMessageState ?? self.currentMessageState,
+            messagesInQueue: messagesInQueue ?? self.messagesInQueue,
+            shownMessageQueueIds: shownMessageQueueIds ?? self.shownMessageQueueIds
+        )
+    }
+
+    static func == (lhs: InAppMessageState, rhs: InAppMessageState) -> Bool {
```

Comment:
"optional", can be an extenstion method, keeps the class lighter.

Reply:
I don't disagree and okay either way. But since it's all related and I think keeping it in the class makes it easier to match, let's stick to that approach.